### PR TITLE
Add PSP pre-checks for unrestricted permissions

### PR DESCRIPTION
**Pull request by liggitt**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6097)

### CODE REVIEW

File: docs/concepts/policy/pod-security-policy.md [hunk 1]
```diff
@@ -207,37 +207,58 @@ podsecuritypolicy "permissive" deleted
...
+
+Check all authenticated users (should answer `no`):
+```shell
+kubectl auth can-i use podsecuritypolicies --as=any-random-user --as-group=system:authenticated
+```
+
+Check all service accounts (should answer `no`):
+```shell
+kubectl auth can-i use podsecuritypolicies --as=system:serviceaccount:any-random-ns:any-random-sa
+```
```
Comment (tengqm) [inline docs/concepts/policy/pod-security-policy.md]: Are "any-random-user", "any-random-ns", "any-random-sa" builtin strings for check? Or we can use `--as=system:serviceaccount:whatever-i-dont-care` as an example? Some clarifications would be helpful.

File: docs/concepts/policy/pod-security-policy.md [hunk 2]
```diff
@@ -207,37 +207,58 @@ podsecuritypolicy "permissive" deleted
...
+
+If any of those checks answer `yes`, it means that the associated set of users has been authorized
+to use any defined PodSecurityPolicy. Locate and remove that permission before continuing.
```
Comment (tengqm) [inline docs/concepts/policy/pod-security-policy.md]: I think the users may anticipate one example or two on "locate and remove", or at least a pointer to these operations.

File: docs/concepts/policy/pod-security-policy.md [hunk 3]
```diff
@@ -207,37 +207,58 @@ podsecuritypolicy "permissive" deleted
...
 
 PodSecurityPolicy authorization uses the union of all policies available to the
-user creating the pod and
-[the service account specified on the pod](/docs/tasks/configure-pod-container/configure-service-account/).
+user creating the pod and [the service account specified on the pod](/docs/tasks/configure-pod-container/configure-service-account/).
+This means that if the control loops started by the [Controller Manager](/docs/admin/kube-controller-manager/)
```
Comment (tengqm) [inline docs/concepts/policy/pod-security-policy.md]: > control loops started by 

This reads a little bit too much about the implementation details. Users don't have control how each control loop is started, IIRC. So ... I'd vote for the original description -- don't start kcm using a superuser account or something like that.

File: docs/concepts/policy/pod-security-policy.md [hunk 4]
```diff
@@ -207,37 +207,58 @@ podsecuritypolicy "permissive" deleted
...
 
 In Kubernetes 1.5 and newer, you can use PodSecurityPolicy to control access to
 privileged containers based on user role and groups. Access to different
 PodSecurityPolicy objects can be controlled via authorization.
 
-Note that [Controller Manager](/docs/admin/kube-controller-manager/) must be run
-against [the secured API port](/docs/admin/accessing-the-api/), and must not
-have superuser permissions. Otherwise requests would bypass authentication and
-authorization modules, all PodSecurityPolicy objects would be allowed,
-and user will be able to create privileged containers.
```
[the remark this answers is not in the document]
Comment (tengqm) [reply docs/concepts/policy/pod-security-policy.md]: Alright. Saw the text that comes later.

File: docs/concepts/policy/pod-security-policy.md [hunk 5]
```diff
@@ -207,37 +207,58 @@ podsecuritypolicy "permissive" deleted
...
+* configured to make requests to the [the secured API port](/docs/admin/accessing-the-api/)
+* running with `--use-service-account-credentials=true` to start each control loop with a restricted credential
 
 For pods created on behalf of a user, in most cases by Controller Manager,
 access should be given to the service account specified on the pod spec
 template. Examples of resources that create pods on behalf of a user are
 Deployments, ReplicaSets, etc.
 
+Access to given PSP policies for a user will be effective only when creating
+Pods directly.
```
Comment (tengqm) [inline docs/concepts/policy/pod-security-policy.md]: Some more elaboration would be helpful.

File: docs/concepts/policy/pod-security-policy.md (hunk 1 above)
Comment (php-coder) [reply docs/concepts/policy/pod-security-policy.md]: > Are "any-random-user", "any-random-ns", "any-random-sa" builtin strings for check?

No, it's supposed that a user will substitute them by a real values.

Comment (liggitt) [reply docs/concepts/policy/pod-security-policy.md]: > No, it's supposed that a user will substitute them by a real values.

Not really, the commands are meant to be run as-is to test access by the named groups

File: docs/concepts/policy/pod-security-policy.md [hunk 6]
```diff
@@ -207,37 +207,58 @@ podsecuritypolicy "permissive" deleted
...
 privileged containers based on user role and groups. Access to different
 PodSecurityPolicy objects can be controlled via authorization.
 
-Note that [Controller Manager](/docs/admin/kube-controller-manager/) must be run
-against [the secured API port](/docs/admin/accessing-the-api/), and must not
-have superuser permissions. Otherwise requests would bypass authentication and
-authorization modules, all PodSecurityPolicy objects would be allowed,
-and user will be able to create privileged containers.
+Before beginning to grant access to specific policies, ensure users and service accounts have not been given global access
+by running the following commands as a privileged user.
```
Comment (php-coder) [inline docs/concepts/policy/pod-security-policy.md]: I'm afraid that some users would misunderstand it and will run this command under `root` user instead. Could you point them that "privileged user" means "a user with `cluster-admin` role"?

File: docs/concepts/policy/pod-security-policy.md (hunk 1 above)
Comment (php-coder) [reply docs/concepts/policy/pod-security-policy.md]: So @tengqm was right that "some clarifications would be helpful".

I don't understand, how the command `kubectl auth can-i use podsecuritypolicies --as=system:serviceaccount:any-random-ns:any-random-sa` will "check all service accounts" then.

**Comment by heckj** [comment]
:wave: @liggitt  - the PR's been pending some updates, wanted to check in with you - also now needs a rebase as well

**Comment by chenopis** [comment]
This is too stale. Feel free to reopen and rebase.

**Pull request closed without merging**